### PR TITLE
fix(react-email): ESM Issue with `await import` of the compiled email templates

### PR DESCRIPTION
**Pull request by AnnatarHe**
## What was the issue?

Errors like the following when running `email export`:

```bash
TypeError: component.default is not a function
    at /Users/app/mailbox/node_modules/.pnpm/react-email@2.0.0_eslint@8.56.0/node_modules/react-email/cli/index.js:778:75
    at step (/Users/app/mailbox/node_modules/.pnpm/react-email@2.0.0_eslint@8.56.0/node_modules/react-email/cli/index.js:186:23)
    at Object.next (/Users/app/mailbox/node_modules/.pnpm/react-email@2.0.0_eslint@8.56.0/node_modules/react-email/cli/index.js:127:20)
    at asyncGeneratorStep (/Users/app/mailbox/node_modules/.pnpm/react-email@2.0.0_eslint@8.56.0/node_modules/react-email/cli/index.js:12:28)
    at _next (/Users/app/mailbox/node_modules/.pnpm/react-email@2.0.0_eslint@8.56.0/node_modules/react-email/cli/index.js:30:17)
```

This was happening because of a `await import` of the esbuild CJS bundled version of the email templates which
had flaky exports because of CJS and ESM incompatibility. The best solution was to use a `require` in the place of a `await import` or a `.default.default` which would be flaky as well.

## How can I test to make sure it's fixed?

1. Run `npx tsx ../../packages/react-email/src/cli/index.ts export` inside of `./apps/demo`
2. Verify that it doesn't error with `TypeError: component.default is not a function`

### CODE REVIEW

File: packages/react-email/src/cli/commands/export.ts
Comment (gabrielmfern) [inline packages/react-email/src/cli/commands/export.ts:65]: Replace the import with a require and I think this will remove the need to do `.default.default`

**Review by gabrielmfern — CHANGES_REQUESTED** [review]

**Review by gabrielmfern — APPROVED** [review]

**Comment by AnnatarHe** [comment]
Hi @gabrielmfern , thanks for your change.

what should I do now? I just want to use the CLI to build my email project 😂

**Comment by gabrielmfern** [comment]
@AnnatarHe Just wait out for a bit until @bukinoshita can review it, I can't approve it anymore since I made changes 😆

**Comment by tim3w4rp** [comment]
Nice. Thank you @gabrielmfern.

@bukinoshita Any update on this? Sorry for pushing you like this but it's blocking me at work. Otherwise, I'd have to either go the separate repo route or use an alternative.

**Comment by gabrielmfern** [comment]
@tim3w4rp You can always use a patch to not be blocked.

**Comment by lightstrike** [comment]
@gabrielmfern Waiting for this to be merged as well to incorporate into a project using Resend. Is there a recommended patching approach in the meantime? I tried the `yarn add 'AnnatarHe/react-email#canary/packages/react-email'` approach through [GitPkg](https://gitpkg.vercel.app/) but it didn't work as expected.

**Comment by kdoroszewicz** [comment]
> @gabrielmfern Waiting for this to be merged as well to incorporate into a project using Resend. Is there a recommended patching approach in the meantime? I tried the `yarn add 'AnnatarHe/react-email#canary/packages/react-email'` approach through [GitPkg](https://gitpkg.vercel.app/) but it didn't work as expected.

You can use patch-package utility to do this. I cloned this repo, installed dependencies, applied changes from this PR and then rebuilt the library to get proper bundles with changes from this PR.  Once you have the file (index.js) you can place it in you projects node_modules and then run patch-package like described in it's readme.

**Review by vcapretz — APPROVED** [review]

**Pull request merged**
